### PR TITLE
Loading offer list

### PR DESCRIPTION
**Pull request by mitia-anah**

### CODE REVIEW

File: src/pages/Home.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import { useAppDispatch } from '../redux/hooks'
+import { useSelector } from 'react-redux'
+import { styled } from '@storybook/theming'
+import { useState } from 'react'
+import { IconSize, Loading } from '../components/Loading/Loading'
+
+export const Home = () => {
+  const dispatch = useAppDispatch()
+  const offers = useSelector(adsSelector)
+  const [isLoading, setIsLoading] = useState<boolean>(false)
```
Comment (lightme-fan) [inline src/pages/Home.tsx:17]: I think it is a good idea to use the status from the fetch ads to set the loading.

File: src/pages/Home.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  console.log(offers)
+
+  useEffect(() => {
+    setTimeout(() => {
```
Comment (lightme-fan) [inline src/pages/Home.tsx:21]: I think you don't need setTimeout here.

File: src/pages/HomePage.tsx [hunk 3]
```diff
@@ -4,25 +4,49 @@ import {
...
-import { IntroContent } from '../components/IntroContent/IntroContent'
+import { styled } from '@storybook/theming'
+import { useState } from 'react'
+import { IconSize, Loading } from '../components/Loading/Loading'
 
 export const HomePage = () => {
   const dispatch = useAppDispatch()
   const offers = useSelector(adsSelector)
+  const [isLoading, setIsLoading] = useState<boolean>(false)
+  console.log(offers)
```
Comment (lightme-fan) [inline src/pages/HomePage.tsx:18]: Remove this console.log.

File: src/pages/Home.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  return (
+    <HomeContainer>
+      {!isLoading ? (
+        <LoadingContainer>
+          <Loading size={IconSize.xl} />
+        </LoadingContainer>
+      ) : (
+        <ListOfOffers offers={offers} />
+      )}
+    </HomeContainer>
```
Comment (Tiaraha-funny) [inline src/pages/Home.tsx:40]: I think we can follow what Anj suggested me that we can just do like this here:

useEffect(() => {
    dispatch(fetchAds([]))
  }, [dispatch])

  return (
   <HomeContainer>
      {!isLoading ? (
        <LoadingContainer>
          <Loading size={IconSize.xl} />
        </LoadingContainer>
      ) : (
        <OffersList offers={offers} />
      )}
    </HomeContainer>
  )
  
  => It means, you don't have to have "ListOfOffers" but instead, you can just call your "OffersList" in you render component.

**Review by lightme-fan — APPROVED** [review]

**Review by Tiaraha-funny — APPROVED** [review]

File: src/pages/HomePage.tsx [hunk 5]
```diff
@@ -1,45 +1,33 @@
...
     <HomeContainer>
-      <IntroContent />
-      <Categories
-        categories={categories}
-        primary={true}
-        selectCategory={() => alert('Selected category')}
-      />
-      <hr />
-      <ListOfOffers offers={offers} />
+      {!isLoading ? (
```
Comment (andrzej-onja) [inline src/pages/HomePage.tsx:24]: offersStatus !== loading

Comment (andrzej-onja) [reply src/pages/HomePage.tsx:24]: in adsSlice
```
export const selectAds = (state: RootState) => state.ads.ads
export const selectStatus = (state: RootState) => state.ads.status
export const adsSelector = createSelector<RootState, any[], any[]>(
  selectAds,
  (ads) => ads
)
export const adsStatusSelector = createSelector<RootState, string, string>(
  selectStatus,
  (status) => status
)

export default adsSlice.reducer
```

File: src/pages/HomePage.tsx [hunk 6]
```diff
@@ -1,45 +1,33 @@
...
+  const [isLoading, setIsLoading] = useState<boolean>(false)
 
-  // Add disptach to useEffect
   useEffect(() => {
-    dispatch(fetchAds([]))
-    dispatch(fetchCategories([]))
+    setTimeout(() => {
+      setIsLoading((item) => !item)
+      dispatch(fetchAds([]))
+    }, 1000)
```
[the remark this answers is not in the document]
Comment (andrzej-onja) [reply src/pages/HomePage.tsx:19]: remove timeout

File: src/graphqlHelper.ts [hunk 7]
```diff
@@ -15,6 +15,7 @@ export const getAdsQuery = () => {
 export const getCategoriesQuery = () => {
   return `{
           categories{title, id, description}
+          ads{title, createdAt, id, username,comments{body, id, username}, photos{url, isPrimary, info}}
```
Comment (andrzej-onja) [inline src/graphqlHelper.ts:18]: this should be a separate query

Comment (andrzej-onja) [reply src/graphqlHelper.ts:18]: remove

File: src/pages/HomePage.tsx [hunk 8]
```diff
@@ -1,45 +1,34 @@
...
     <HomeContainer>
-      <IntroContent />
-      <Categories
-        categories={categories}
-        primary={true}
-        selectCategory={() => alert('Selected category')}
-      />
-      <hr />
-      <ListOfOffers offers={offers} />
+      {offerStatus === 'loading' ? (
```
Comment (andrzej-onja) [inline src/pages/HomePage.tsx:25]: create in folder src/constants/index.ts
export enum FETCH_STATUS {
LOADING= 'loading',
IDLE='idle'

}

File: src/redux/slices/adsSlice.tsx [hunk 9]
```diff
@@ -60,7 +60,7 @@ export const adsSlice = createSlice({
     },
     //TODO remove used only as an example
     decrement: (state) => {
-      // console.log(state.value, 'dec')
+      // console.log(stat offerStatus === 'loading'e.value, 'dec')
```
Comment (andrzej-onja) [inline src/redux/slices/adsSlice.tsx:63]: use FETCH_STATUS instead of status strings

File: src/redux/slices/adsSlice.tsx [hunk 10]
```diff
@@ -104,13 +81,16 @@ export const adsSlice = createSlice({
...
+export const selectStatus = (state: RootState) => state.ads.status
+
 export const adsSelector = createSelector<RootState, any[], any[]>(
   selectAds,
   (ads) => ads
 )
+export const adsStatusSelector = createSelector<RootState, string, string>(
+  selectStatus,
+  (status) => status
+)
```
Comment (andrzej-onja) [inline src/redux/slices/adsSlice.tsx:95]: in this file look for `state.status` and instead of strings use the new enum

**Review by andrzej-onja — APPROVED** [review]

**Pull request merged**